File: backend/Model/UserProfileInfo.cs
using System.Collections.Generic;

namespace tradeus.Model
{
    public class UserProfileInfo
    {
        public int UserId { get; set; }
        public string Username { get; set; }
        public int NumberOfPosts { get; set; }
        public int NumberOfFollowers { get; set; }
        public int NumberOfFollowing { get; set; }
        public List<PostDetailSimple> postDetailSimples { get; set; }
        public bool IsFollowing { get; set; }
    }
}

File: frontend/src/models/UserProfileInfo.ts
import PostDetailSimple from "./PostDetailSimple";

export default class UserProfileInfo {
    userId = 0;
    numberOfPosts = 0;
    numberOfFollowers = 0;
    numberOfFollowing = 0;
    username = "";
    postDetailSimples = [] as PostDetailSimple[];
    isFollowing = false;
}
